web/src/components/molecules/NavBarItem/index.tsx

```diff
@@ -0,0 +1,72 @@
+'use client';
+
+import { SvgIconComponent } from '@mui/icons-material';
+import { Box, Typography } from '@mui/material';
+import Link from 'next/link';
+import { usePathname } from 'next/navigation';
+import { FC } from 'react';
+
+interface Props {
+  label: string;
+  Icon: SvgIconComponent;
+  active?: boolean;
+  width?: number | string;
+  height?: number | string;
+  paddingX?: number | string;
+  gap?: number;
+  linkTo?: string;
+}
+
+const NavBarItem: FC<Props> = ({
+  label,
+  Icon,
+  active = false,
+  width = 200,
+  height = 40,
+  paddingX = 2,
+  gap = 1,
+  linkTo = '/'
+}) => {
+  const pathname = usePathname();
+  active = pathname === linkTo ? true : false;
```

Comment:
better if active is immutable, pathname === linkTo should be handled above this component. This component will just receive true or false.